airflow/contrib/operators/aws_athena_operator.py

```diff
@@ -74,7 +76,16 @@ def execute(self, context):
         self.result_configuration['OutputLocation'] = self.output_location
         self.query_execution_id = self.hook.run_query(self.query, self.query_execution_context,
                                                       self.result_configuration, self.client_request_token)
-        self.hook.poll_query_status(self.query_execution_id)
+        query_status = self.hook.poll_query_status(self.query_execution_id, self.max_tries)
+
+        if not query_status or query_status in AWSAthenaHook.FAILURE_STATES:
+            raise Exception(
+                'Athena job failed. Final state is {}, query_execution_id is {}.'
+                .format(query_status, self.query_execution_id))
+        elif query_status in AWSAthenaHook.INTERMEDIATE_STATES:
```

Comment:
I guess it should be like below?
```python
if query_status in AWSAthenaHook.FAILURE_STATES:
    raise Exception("...")
elif not query_status or query_status in AWSAthenaHook.INTERMEDIATE_STATES:
    raise Exception("...")
```

Reply:
Yes. I updated this one. Thx for noticing it.

Reply:
Hi @bryanyang0528 seems you only fixed for the `.format()` method, but the `if...elif` question is not addressed yet.

Reply:
@XD-DENG  According to the source code of `query_status`:
```
if query_state is None:
                self.log.info('Trial {try_number}: Invalid query state. Retrying again'.format(
                    try_number=try_number))
```
This query might something wrong so that this query is not sent to the queue of Athena, so I think it should show `failed` if `query_status` is None. Does it make any sense or there is somthing I missing?

Reply:
Hi @bryanyang0528 , if you check https://github.com/apache/airflow/blob/4655c3f2bbd6dbb442a9c8482559748bd9db0bd7/airflow/contrib/hooks/aws_athena_hook.py#L123-L140

You will notice that `query_state is None` or `query_state in self.INTERMEDIATE_STATES` would not result in `break`. Instead, `poll_query_status()` will only end with `query_state is None` or `query_state in self.INTERMEDIATE_STATES` when `max_tries` is reached. It may be a too strong assumption to say "`query_status` is `None` means `failed`".

On the other hand, `else:` (including `FAILURE_STATES`) cause an explicit `break`, which is for sure `failed`.

Hope this clarifies.

Reply:
@XD-DENG Thank you for the detailed explanation. It's very clear and helpful.